common/beerocks/bwl/shared/nl80211_client_impl.cpp

```diff
@@ -17,13 +21,276 @@
+
+                        if (band.ht_capability & BIT(1)) {
+                            ctx.width_40 = true;
+                        }
+                    } else {
+                        LOG(DEBUG) << "NL80211_BAND_ATTR_HT_CAPA attribute is missing";
+                    }
+
+                    if (tb_band[NL80211_BAND_ATTR_HT_MCS_SET]) {
+                        size_t expected_length = sizeof(band.ht_mcs_set);
+                        size_t actual_length   = nla_len(tb_band[NL80211_BAND_ATTR_HT_MCS_SET]);
+
+                        if (actual_length <= expected_length) {
+                            memcpy(band.ht_mcs_set, nla_data(tb_band[NL80211_BAND_ATTR_HT_MCS_SET]),
+                                   actual_length);
+                        } else {
+                            LOG(DEBUG) << "Invalid length for NL80211_BAND_ATTR_HT_MCS_SET "
+                                          "attribute. Expected length: "
+                                       << expected_length << ", actual length: " << actual_length;
+                        }
+                    } else {
+                        LOG(DEBUG) << "NL80211_BAND_ATTR_VHT_MCS_SET attribute is missing";
+                    }
+
+                    if (tb_band[NL80211_BAND_ATTR_VHT_CAPA]) {
+                        band.vht_capability = nla_get_u32(tb_band[NL80211_BAND_ATTR_VHT_CAPA]);
+
+                        ctx.width_80 = true;
+
+                        switch ((band.vht_capability >> 2) & 3) {
+                        case 2:
+                            /* width_80p80 = true; */
+                            /* fall through */
+                        case 1:
+                            ctx.width_160 = true;
+                            break;
+                        }
+                    } else {
+                        LOG(DEBUG) << "NL80211_BAND_ATTR_VHT_CAPA attribute is missing";
+                    }
+
+                    if (tb_band[NL80211_BAND_ATTR_VHT_MCS_SET]) {
+                        size_t expected_length = sizeof(band.vht_mcs_set);
+                        size_t actual_length   = nla_len(tb_band[NL80211_BAND_ATTR_VHT_MCS_SET]);
+
+                        if (actual_length <= expected_length) {
+                            memcpy(band.vht_mcs_set,
+                                   nla_data(tb_band[NL80211_BAND_ATTR_VHT_MCS_SET]), actual_length);
+                        } else {
+                            LOG(DEBUG) << "Invalid length for NL80211_BAND_ATTR_VHT_MCS_SET "
+                                          "attribute. Expected length: "
+                                       << expected_length << ", actual length: " << actual_length;
+                        }
+                    } else {
+                        LOG(DEBUG) << "NL80211_BAND_ATTR_VHT_MCS_SET attribute is missing";
+                    }
+
+                    if (tb_band[NL80211_BAND_ATTR_FREQS]) {
```

Comment:
I think we can treat it as an error (and return) if this is missing. That saves us some indentation depth below.

Reply:
Done